src/client.rs

```diff
@@ -7,7 +7,7 @@ use log::{error, trace, warn, Level as LogLevel, Log, Record};
 use serde_json::to_string;
 use std::{
     cell::RefCell,
-    collections::{HashMap, LinkedList},
+    collections::{HashMap, VecDeque},
```

Comment:
Ah, moved to `VecDeque` I see. Any particular reason?

Reply:
Yeah, read a bunch of articles all of which said to use VecDeque because of Rust-specific performance improvements due to contiguous memory.  Plus, quicker to iterate from the back (because it's just contiguous mem checks), although I would venture it would be hard to see the actual difference with the small sizes of collections we would have :)